src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/CustomerGroupServiceImplIT.java

```diff
@@ -0,0 +1,81 @@
+package com.commercetools.sync.integration.sdk2.services.impl;
+
+import static com.commercetools.sync.integration.sdk2.commons.utils.CustomerGroupITUtils.createCustomerGroup;
+import static com.commercetools.sync.integration.sdk2.commons.utils.CustomerGroupITUtils.deleteCustomerGroups;
+import static com.commercetools.sync.integration.sdk2.commons.utils.TestClientUtils.CTP_TARGET_CLIENT;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.commercetools.api.models.customer_group.CustomerGroup;
+import com.commercetools.sync.sdk2.products.ProductSyncOptions;
+import com.commercetools.sync.sdk2.products.ProductSyncOptionsBuilder;
+import com.commercetools.sync.sdk2.services.CustomerGroupService;
+import com.commercetools.sync.sdk2.services.impl.CustomerGroupServiceImpl;
+import java.util.ArrayList;
+import java.util.Optional;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CustomerGroupServiceImplIT {
+  private CustomerGroupService customerGroupService;
+  private static final String CUSTOMER_GROUP_KEY = "customerGroup_key";
+  private static final String CUSTOMER_GROUP_NAME = "customerGroup_name";
+
+  private CustomerGroup oldCustomerGroup;
+  private ArrayList<String> warnings;
+
+  /**
+   * Deletes customer group from the target CTP projects, then it populates the project with test
+   * data.
+   */
+  @BeforeEach
+  void setup() {
+    deleteCustomerGroups(CTP_TARGET_CLIENT);
+    warnings = new ArrayList<>();
+    oldCustomerGroup =
+        createCustomerGroup(CTP_TARGET_CLIENT, CUSTOMER_GROUP_NAME, CUSTOMER_GROUP_KEY);
```

Comment:
Please use `ensureCustomerGroup()` instead and try not to delete custom groups after every test suit run. If we could eliminate these deletions, tests would become more stable.